### ISSUE TITLE
rescale Date (and POSIXct) objects?

### ISSUE THREAD

**Issue by zeehio** [hadley/scales#74]
Hi, 

As of today it is not possible to rescale dates (and times) using `scales::rescale`. Is there a particular reason? I am puzzled by this, and I think that I must be missing something, because the problem seems fairly simple to tackle (using `as.numeric` somewhere). I would like to contribute with a clean solution, but I am afraid I am being naive on the solution. Any advice? (ggplot2::geom_segment [could benefit](https://github.com/hadley/ggplot2/issues/1048) from this)

```
days <- as.Date(c("2016-03-01", "2016-03-02", "2016-03-03"))
scales::rescale(days[2], from = c(days[1], days[3])) == 0.5

datetimes <- as.POSIXct(c("2016-03-01 08:05:00",
                          "2016-03-01 08:10:00",
                          "2016-03-01 08:15:00"),
                        format = "%Y-%m-%d %H:%M:%S")
testthat::expect_equal(scales::rescale(datetimes[2],
                       from = c(datetimes[1], datetimes[3])), 0.5)
```

I get

```
Error in Math.Date(x) : abs not defined for "Date" objects
```

**Comment by hadley** [comment]
I think coercing back and forth between numerics is probably the right thing to do.

**Comment by zeehio** [comment]
I just rebased the pull request on the latest master. Is there anything else I can do to get this merged?

Thank you for your time

**Comment by zeehio** [comment]
This is already merged in #75, so I will close it
